packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -1186,20 +1190,12 @@ export function TransferPanel() {
     customFeeTokenL1BalanceFloat
   ])
 
-  const gasSummary = useGasSummary(
-    amountBigNumber,
-    selectedToken,
-    shouldRunGasEstimation
-  )
+  useGasSummary(amountBigNumber, selectedToken, shouldRunGasEstimation)
 
-  const { transferReady, errorMessage } = useTransferReadiness({
-    amount,
-    gasSummary
-  })
+  const { gasSummary } = useGasSummaryStore()
```

Comment:
If I understand that correctly:
- `useGasSummary` trigger the fetch
- `useGasSummaryStore` returns the result

Can we wrap it within one hook? Something like `const { gasSummary } = useGasSummaryState(amountBigNumber, selectedToken, shouldRunGasEstimation)` which trigger the fetch, and also return the result?